File: AndroidBasic/app/src/main/java/com/ylg/androidbasic/RecyclerAdapter.java
package com.ylg.androidbasic;

import android.app.Activity;
import android.content.Context;
import android.content.Intent;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.ImageButton;
import android.widget.TextView;

import java.util.List;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

public class RecyclerAdapter extends RecyclerView.Adapter<RecyclerAdapter.ViewHolder> {

    String TAG = RecyclerAdapter.class.getName();

    private LayoutInflater inflater;
    private List<String> activitys;
    private Context context;

    public RecyclerAdapter(Context context) {
        this.context = context;
        inflater = LayoutInflater.from(context);
        activitys = ((MyApplication) context.getApplicationContext()).getActivitys();
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = inflater.inflate(R.layout.item_recycler_layout, parent, false);
        //view.setBackgroundColor(Color.RED);
        ViewHolder viewHolder = new ViewHolder(view);
        return viewHolder;
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        try {
            //Class clazz = getClassLoader().loadClass("com.ylg.androidbasic.TestActivity");
            Class clazz = Class.forName(activitys.get(position));
            //Log.i(TAG, "Title = " + clazz.getDeclaredField("title1").get(clazz.newInstance()));
            holder.btn.setText(clazz.getDeclaredField("title").get(clazz.newInstance()).toString());
        } catch (Exception e) {
            Log.i(TAG, "RecyclerAdapter onBindViewHolder {position=" + position + ", "
                    + activitys.get(position) + "} error: " + e.getMessage());
        }
        holder.itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                try {
                    //Class clazz = getClassLoader().loadClass("com.ylg.androidbasic.TestActivity");
                    Class clazz = Class.forName(activitys.get(position));
                    //Activity clazz1 =  (Activity)clazz.newInstance();
                    //Log.i(TAG, "Title = " + clazz.getDeclaredField("title1").get(clazz.newInstance()));
                    Intent intent = new Intent(context, clazz);
                    context.startActivity(intent);
                } catch (Exception e) {
                    Log.i(TAG, "RecyclerAdapter startActivity {position=" + position + ", "
                            + activitys.get(position) + "} error: " + e.getMessage());
                }
            }
        });
    }

    @Override
    public int getItemCount() {
        return activitys.size();
    }

    class ViewHolder extends RecyclerView.ViewHolder {

        public TextView tv;
        public Button btn;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            btn = (Button) itemView;
        }
    }
}

File: MyKotlinStudy/app/src/main/java/com/ylg/mykotlinstudy/RecyclerAdapter.kt
package com.ylg.mykotlinstudy

import android.content.Context
import android.content.Intent
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView

class RecyclerAdapter(private val context: Context) :
    RecyclerView.Adapter<RecyclerAdapter.ViewHolder>() {
    var TAG = RecyclerAdapter::class.java.name
    private val inflater: LayoutInflater
    private val activitys: List<String>
    override fun onCreateViewHolder(
        parent: ViewGroup,
        viewType: Int
    ): ViewHolder {
        val view =
            inflater.inflate(R.layout.item_recycler_layout, parent, false)
        //view.setBackgroundColor(Color.RED);
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        try {
            //Class clazz = getClassLoader().loadClass("com.ylg.androidbasic.TestActivity");
            val clazz = Class.forName(activitys[position])
            //Log.i(TAG, "Title = " + clazz.getDeclaredField("title1").get(clazz.newInstance()));
            //val javaclass = clazz.newInstance().javaClass
            //holder.btn.text = clazz.getDeclaredField("title1").toString()
            //holder.btn.text =  javaclass.getField("title1").get(javaclass.newInstance()).toString()
            holder.btn.text = clazz.getMethod("getTitle1").invoke(clazz.newInstance()).toString()
                //.invoke(clazz.newInstance(),null).toString()

        } catch (e: Exception) {
            Log.i(
                TAG,
                "RecyclerAdapter onBindViewHolder {position=" + position + ", " + activitys[position] + "} error: " + e.message
            )
        }
        holder.itemView.setOnClickListener {
            try {
                //Class clazz = getClassLoader().loadClass("com.ylg.androidbasic.TestActivity");
                val clazz = Class.forName(activitys[position])
                //Activity clazz1 =  (Activity)clazz.newInstance();
                //Log.i(TAG, "Title = " + clazz.getDeclaredField("title1").get(clazz.newInstance()));
                val intent = Intent(context, clazz)
                context.startActivity(intent)
            } catch (e: Exception) {
                Log.i(
                    TAG, "RecyclerAdapter startActivity {position=" + position + ", "
                            + activitys[position] + "} error: " + e.message
                )
            }
        }
    }

    override fun getItemCount(): Int {
        return activitys.size
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var tv: TextView? = null
        var btn: Button

        init {
            btn = itemView as Button
        }
    }

    init {
        inflater = LayoutInflater.from(context)
        activitys = (context.applicationContext as MyApplication).activitys
    }
}